src/views/htmlcontent/src/js/components/app.component.ts

```diff
@@ -163,9 +173,15 @@ export class AppComponent implements OnInit, AfterViewChecked {
             this.navigateToGrid(this.activeGrid - 1);
         },
         'event.copySelection': () => {
-            let activeGrid = this.activeGrid;
-            let selection = this.slickgrids.toArray()[activeGrid].getSelectedRanges();
-            this.dataService.copyResults(selection, this.renderedDataSets[activeGrid].batchId, this.renderedDataSets[activeGrid].resultId);
+            let range: IRange = this.getSelectedRangeUnderMessages();
+            let messageText = range ? range.text() : '';
+            if (messageText.length > 0) {
+                this.executeCopy(messageText);
+            } else {
```

Comment:
I think the current behavior is fine but not ideal. It's hard to make a comparison with SSDT and SSMS because they put messages and resultsets in different tabs.